src/go/rpk/pkg/config/config_test.go

```diff
@@ -426,103 +301,54 @@ func TestWrite(t *testing.T) {
-  - address: 0.0.0.0
-    port: 9092
-  node_id: 0
-  rpc_server:
-    address: 0.0.0.0
-    port: 33145
-  seed_servers:
-  - host:
-      address: 127.0.0.1
-      port: 33145
-  - host:
-      address: 127.0.0.1
-      port: 33146
+    data_directory: /var/lib/redpanda/data
+    node_id: 0
+    seed_servers:
+        - host:
+            address: 127.0.0.1
+            port: 33145
+        - host:
+            address: 127.0.0.1
+            port: 33146
+    rpc_server:
+        address: 0.0.0.0
+        port: 33145
+    kafka_api:
+        - address: 0.0.0.0
+          port: 9092
+    admin:
+        - address: 0.0.0.0
+          port: 9644
+    developer_mode: false
 rpk:
-  coredump_dir: /var/lib/redpanda/coredumps
-  enable_memory_locking: true
-  enable_usage_stats: true
-  overprovisioned: false
-  tune_aio_events: true
-  tune_ballast_file: true
-  tune_clocksource: true
-  tune_coredump: true
-  tune_cpu: true
-  tune_disk_irq: true
-  tune_disk_nomerges: true
-  tune_disk_scheduler: true
-  tune_disk_write_cache: true
-  tune_fstrim: true
-  tune_network: true
-  tune_swappiness: true
-  tune_transparent_hugepages: true
-  well_known_io: vendor:vm:storage
-schema_registry: {}
-`,
-		},
-		{
-			name:    "it should write the default values if redpanda.kafka_api is missing",
-			conf:    getValidConfig,
-			wantErr: false,
-			expected: `config_file: /etc/redpanda/redpanda.yaml
+    enable_usage_stats: true
+    tune_network: true
+    tune_disk_scheduler: true
+    tune_disk_nomerges: true
+    tune_disk_write_cache: true
+    tune_disk_irq: true
+    tune_fstrim: true
+    tune_cpu: true
+    tune_aio_events: true
+    tune_clocksource: true
+    tune_swappiness: true
+    tune_transparent_hugepages: true
+    enable_memory_locking: true
+    tune_coredump: true
+    coredump_dir: /var/lib/redpanda/coredumps
+    tune_ballast_file: true
+    well_known_io: vendor:vm:storage
+    overprovisioned: false
 pandaproxy: {}
-redpanda:
-  admin:
-  - address: 0.0.0.0
-    port: 9644
-  data_directory: /var/lib/redpanda/data
-  developer_mode: false
-  kafka_api:
-  - address: 0.0.0.0
-    port: 9092
-  node_id: 0
-  rpc_server:
-    address: 0.0.0.0
-    port: 33145
-  seed_servers:
-  - host:
-      address: 127.0.0.1
-      port: 33145
-  - host:
-      address: 127.0.0.1
-      port: 33146
-rpk:
-  coredump_dir: /var/lib/redpanda/coredumps
-  enable_memory_locking: true
-  enable_usage_stats: true
-  overprovisioned: false
-  tune_aio_events: true
-  tune_ballast_file: true
-  tune_clocksource: true
-  tune_coredump: true
-  tune_cpu: true
-  tune_disk_irq: true
-  tune_disk_nomerges: true
-  tune_disk_scheduler: true
-  tune_disk_write_cache: true
-  tune_fstrim: true
-  tune_network: true
-  tune_swappiness: true
-  tune_transparent_hugepages: true
-  well_known_io: vendor:vm:storage
 schema_registry: {}
 `,
 		},
 		{
-			name: "shall write a valid config file without advertised_kafka_api",
+			name: "write additional values",
```

Comment:
How'd this test change? hard to follow in the diff

Reply:
We want to test that the new config's method Write actually writes what is loaded in memory (config) to a file.

The changes here are to show that we can:

1. Write defaults
2. Write empty structs (if any)
3. Write unrecognized values (Other Map)

The other tests were testing rewrites but now we have separated logic: Load and Write.